library/core/src/pin.rs

```diff
@@ -387,17 +884,122 @@ use crate::hash::{Hash, Hasher};
 use crate::marker::{Sized, Unpin};
 use crate::ops::{CoerceUnsized, Deref, DerefMut, DispatchFromDyn, Receiver};
 
-/// A pinned pointer.
+#[allow(unused_imports)]
+use crate::{
+    cell::{RefCell, UnsafeCell},
+    future::Future,
+    marker::PhantomPinned,
+    mem, ptr,
+};
+
+/// A pointer which pins its pointee in place.
+///
+/// [`Pin`] is a wrapper around some kind of pointer `Ptr` which makes that pointer "pin" its
```

Comment:
Perhaps, yeah. I do think that `Ptr` is much more clear to someone unfamiliar with pinning. `P` sounds like it could be the thing being pinned! But it is unfortunate to have a split....

Not sure what the best solution is.

Reply:
FWIW, I am _very strongly_ in favor of `Ptr`; seeing the docs using it (is a sight for sore eyes 😍 and) does improve a lot the _otherwise uncommon_ notion that this wrapper acts through pointer indirection rather than wrapping the "pinned thing" itself

Reply:
FWIW I would also be in favor of renaming it in the actual code as well, though that may be a bit more of a controversial change... it has my +1 though. I guess I'll leave this in discussion for now and wait for further feedback and then we can decide where to go.

Reply:
I've changed the code and all references in the docs from `Pin<P>` to `Pin<Ptr>` in cd13f5c ... :)

Reply:
It's more common for std to use single-letter generics but I believe this is a case where using the "full" version is justified.